### ISSUE TITLE
conversion tools

### ISSUE THREAD

**Issue by Remi-Gau** [ReproNim/reproschema-py#3]
hey @satra 

in case some of the ugly python code I have been creating to create the cobidas schema is useful, feel free to reuse it to recycle and improve.

it is here: https://github.com/Remi-Gau/COBIDAS_chckls/tree/master/conversion_scripts

**Comment by satra** [comment]
thanks @remi - this should be helpful for the python version. i spoke to @sanuann yesterday and she plans to update the javascript converter as that may still be useful in some javascript tabular ui.

**Comment by Remi-Gau** [comment]
I also think that having some "tools" in both languages might definitely help with adoption. :smile:

**Comment by satra** [comment]
@Remi-Gau - when you get a chance could you please send a PR adding your conversion script, so that we know you contributed it. i can then refactor it to do the conversion from google spreadsheets/redcap csv's.

**Comment by Remi-Gau** [comment]
@satra 
Quick question: what would be a sensible "testing" strategy for the code that should help with schema creation?

Would the idea be to have a sample of test.jsonld files that we check the output of the test against?

**Comment by satra** [comment]
Yes. Having a set of files/folders that can be produced by the generation code would be a set of tests. 

To be comprehensive, we would want a set of crud (create, read, update, delete) tests

**Comment by Remi-Gau** [comment]
will have to get familiar with crud then because the only crud I knew till now was that of [Paul Mheel](https://journals.sagepub.com/doi/full/10.1177/2515245920917961)

OK so apparently this is completely different from setup and teardown within a test, these are the different types of behavior we would like to create test for, is that correct?

**Comment by satra** [comment]
>  these are the different types of behavior we would like to create test for, is that correct?

yes, if you think of the reproschema-py as kind of an API and reproschema as a DB, then the library needs tests for its operations, which are referred to as CRUD. these would be separate tests, but they may have many variants as well. it is a good lens to think through about the use cases and tests.
